lib/veil/credential_collection/chef_secrets_env.rb

```diff
@@ -0,0 +1,38 @@
+require "veil/credential_collection/base"
+require "json"
+
+module Veil
+  class CredentialCollection
+    class ChefSecretsEnv < Base
+
+      # Create a new ChefSecretsEnv
+      #
+      # @param [Hash] opts
+      #   a hash of options to pass to the constructor
+      def initialize(opts = {})
+        var_name = opts[:var_name] || 'CHEF_SECRETS_DATA'
+
+        @credentials = {}
+        import_credentials_hash(read_env_var(var_name))
+      end
+
+      # Unsupported methods
+      def rotate
+        raise NotImplementedError
+      end
+      alias_method :rotate_credentials, :rotate
```

Comment:
For clarity it might be good to have a base class method like:
```
def unimplemented
    raise NotImplementedError
end
```

And alias each of these methods to `unimplemented`.